backends-clickhouse/src/main/scala/org/apache/spark/rpc/GlutenDriverEndpoint.scala

```diff
@@ -32,16 +32,14 @@ import java.util.concurrent.atomic.AtomicInteger
  * The gluten driver endpoint is responsible for communicating with the executor. Executor will
  * register with the driver when it starts.
  */
-class GlutenDriverEndpoint extends IsolatedRpcEndpoint with Logging {
+class GlutenDriverEndpoint extends IsolatedThreadSafeRpcEndpoint with Logging {
   override val rpcEnv: RpcEnv = SparkEnv.get.rpcEnv
 
   protected val totalRegisteredExecutors = new AtomicInteger(0)
 
   private val driverEndpoint: RpcEndpointRef =
     rpcEnv.setupEndpoint(GlutenRpcConstants.GLUTEN_DRIVER_ENDPOINT_NAME, this)
 
-  // TODO(yuan): get thread cnt from spark context
-  override def threadCount(): Int = 1
```

Comment:
If I recall correctly this piece of code is left to make it compile on different Spark versions, we did not use the thread count here for real work

Reply:
Thanks, I just noticed that IsolatedThreadSafeRpcEndpoint was imported from spark 3.4. I suggest to remove TODO, all we need is just a thread-safe endpoint.